src/app/enterprise.service.ts

```diff
@@ -0,0 +1,30 @@
+/*
+ * SPDX-FileCopyrightText: 2022-present Open Networking Foundation <info@opennetworking.org>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { Injectable } from '@angular/core';
+import { TargetsNames } from '../openapi3/top/level/models';
+import { ApiService as TopLevelApiService } from '../openapi3/top/level/services';
+import { tap } from 'rxjs/operators';
+
+@Injectable({
+    providedIn: 'root',
+})
+export class EnterpriseService {
+    enterprises: TargetsNames;
```

Comment:
I'm not a huge supported of directly accessing class property, maybe making this private/protected and using a `getter` would provide better isolation?

A pattern I found useful in the past was to always call a `load` (or similar) methods, that would implement a cache if needed but always return a `promise/observable`, that way the "client" doesn't need to know whether the data is already loaded or not.

Reply:
I added a getter